components/discord/actions/send-message-with-file/send-message-with-file.mjs

```diff
@@ -38,29 +29,18 @@ export default {
       avatarURL,
       threadID,
       username,
-      fileUrl,
-      filePath,
+      file,
       includeSentViaPipedream,
       suppressNotifications,
     } = this;
 
-    if (!fileUrl && !filePath) {
-      throw new ConfigurationError("This action requires either File URL or File Path. Please enter one or the other above.");
-    }
-
-    const file = fileUrl
-      ? await axios($, {
-        method: "get",
-        url: fileUrl,
-        responseType: "stream",
-      })
-      : fs.createReadStream(filePath);
+    const stream = await getFileStream(file);
```

Comment:
It's a bit of a hack, since a `name` property is not part of the Stream spec, but [`form-data` appears to get `filename` from `.name` if it exists](https://github.com/form-data/form-data/blob/d8d67dc8ac79285154edf7d3f57dbab593b9a146/lib/form_data.js#L229). Maybe this should have been done by the `getFileStream` function.

```suggestion
    const {
      stream, metadata,
    } = await getFileStreamAndMetadata(file);
    stream.name = metadata.name;
```